File: ws00_helloworld/src/pkg02_helloworld_py/pkg02_helloworld_py/helloworld.py
# def main():
#     print('Hi from pkg02_helloworld_py.')


# if __name__ == '__main__':
#     main()

import rclpy
from rclpy.node import Node

class MyNode(Node):
    def __init__(self,node_name):
        super().__init__(node_name)
        self.get_logger().info("hello world")

def main():
    rclpy.init()
    node = MyNode("py_node")
    rclpy.spin(node)
    rclpy.shutdown()



    

        

File: ws00_helloworld/src/pkg01_helloworld_cpp/src/helloworld.cpp
// #include <cstdio>

// int main(int argc, char ** argv)
// {
//   (void) argc;
//   (void) argv;

//   printf("hello world pkg01_helloworld_cpp package\n");
//   return 0;
// }

#include <rclcpp/rclcpp.hpp>

class MyNode: public rclcpp::Node{
  public: 
    MyNode(const std::string &node_name):Node(node_name){
      RCLCPP_INFO(this->get_logger(),"hello world");
    }

};

int main(int argc, char* argv[]){
  rclcpp::init(argc, argv);
  auto node = std::make_shared<MyNode>("cpp_node");
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}
















